### PR TITLE
Fabric8 ui #51

### PR DESCRIPTION
**Pull request by pmuir**
Closes #78 

Fixes linting issues

### CODE REVIEW

File: src/app/home/home.component.html [hunk 1]
```diff
@@ -62,8 +62,8 @@
   </div>
 </div>
 <!-- Create Space modal -->
-<modal #createSpace title="Create a new collaboration space">
-  <modal-content>
-    <space-dialog></space-dialog>
+<modal #createSpace class="chromeless-modal"  >
+  <modal-content class="chromeless-modal-content" >
+      <space-wizard></space-wizard>
```
Comment (joshuawilson) [inline src/app/home/home.component.html]: please remove the extra space

Comment (mitchharpur) [inline src/app/home/home.component.html]: This dialog still has several more iterations ... can we merge and defer to another pull request

File: src/app/home/home.module.ts [hunk 2]
```diff
@@ -1,3 +1,4 @@
+import { SpaceWizardModule } from '../space-wizard/space-wizard.module';
```
Comment (joshuawilson) [inline src/app/home/home.module.ts]: As a style standard the imports are organized with @angular first, external imports, then internal imports. There are 2 ways to order them after that both are alphabetically. By the component or the location of the component. For larger imports, I tend to group by the location and then sort by component. For smaller imports I just sort by component only.

File: src/app/space-wizard/space-wizard.component.html [hunk 3]
```diff
@@ -0,0 +1,191 @@
...
+                <div class="col-sm-8">
+                  <select class="form-control" id="projectType" [(ngModel)]="configurator.project.type" name="projectType">
+                            <option *ngFor="let projectType of configurator.availableProjectTypes" [ngValue]="pt">{{projectType.description}}</option>
+                        </select>
+                </div>
+              </div>
+              <div class="form-group">
+                <label class="control-label col-sm-4">Project Name *</label>
+                <div class="col-sm-8">
+                  <input id="projectName" name="projectName" type="text" class="form-control" placeholder="demo" [(ngModel)]="configurator.project.name"
```
Comment (joshuawilson) [inline src/app/space-wizard/space-wizard.component.html]: duplicate ID with line 148

File: src/app/space-wizard/space-wizard.component.html [hunk 4]
```diff
@@ -0,0 +1,191 @@
...
+  </div>
+  <div class="wizard-step" [class.active]="wizard.activeStep===wizardSteps.quickStart.index">
+    <header>
+      <h3>Define a QuickStart to add to the <strong>BaloonPopGame</strong> space. </h3>
+    </header>
+    <section class="margin-top-15">
+      <div class="container-fluid">
+        <div class="row">
+          <div class="col-xs-12">
+            <form class="form-horizontal">
```
Comment (joshuawilson) [inline src/app/space-wizard/space-wizard.component.html]: I don't know if you are aware but Angular has form controls. I guess we can refactor this later to use them if you are unsure how they work.

File: src/app/space-wizard/space-wizard.component.html [hunk 5]
```diff
@@ -0,0 +1,191 @@
...
+              <div class="form-group has-clear">
+                <label class="control-label col-sm-4">Project type *</label>
+                <div class="col-sm-8">
+                  <select class="form-control" id="projectType" [(ngModel)]="configurator.project.type" name="projectType">
```
Comment (joshuawilson) [inline src/app/space-wizard/space-wizard.component.html]: I think <select> should have the <label> with it, not a level above. If I remember right, in HTML they need to be next to each other to link them. Or you can use "name" attribute, I think.

File: src/app/space-wizard/space-wizard.component.ts [hunk 6]
```diff
@@ -0,0 +1,52 @@
+import { Component, OnInit } from '@angular/core';
+import { Router } from '@angular/router';
+import { DummyService } from '../dummy/dummy.service';
+import { SpaceConfigurator, IWizardSteps, Wizard, WizardNavigator, ProjectType, ProjectInfo, StackInfo, PipelineInfo, PipelineStageInfo, PipelineTaskInfo, PipelineEnvironmentInfo } from '../models/wizard';
```
Comment (joshuawilson) [inline src/app/space-wizard/space-wizard.component.ts]: You have several unused components in this list.

File: src/app/space-wizard/space-wizard.component.ts [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+import { Space } from '../models/space';
+import { ProcessTemplate } from '../models/process-template';
```
Comment (joshuawilson) [inline src/app/space-wizard/space-wizard.component.ts]: This does not appear to be used.

File: src/assets/stylesheets/_icons.scss [hunk 8]
```diff
@@ -0,0 +1,97 @@
...
+  }
+  &.icon-stack-python
+  {
+    background-image: url("#{$icon-path}icon-stack-python.png");
+  }
+  &.icon-stack-maven
+  {
+    background-image: url("#{$icon-path}icon-stack-maven.png");
+    background-size: 110%;
+    background-position-x: 0px;
```
Comment (joshuawilson) [inline src/assets/stylesheets/_icons.scss]: 0 with units

**Comment by joshuawilson** [comment]
This is full of issues but because it is from another PR, it is just easier to fix the problems in a new PR. So I'm merging it as is. :(

File: src/app/home/home.module.ts (hunk 2 above)
Comment (pmuir) [inline src/app/home/home.module.ts]: Unfortunately tools don't seem to recognise this so it's too easy to get it wrong. Maybe we just drop this styleguide until tools can support it?

File: src/app/space-wizard/space-wizard.component.html (hunk 4 above)
Comment (pmuir) [inline src/app/space-wizard/space-wizard.component.html]: Do you mean the ngSubmit functionality? If so, yes, that is useful.

Comment (pmuir) [inline src/app/space-wizard/space-wizard.component.html]: /cc @mitchharpur

File: src/app/space-wizard/space-wizard.component.html (hunk 5 above)
Comment (pmuir) [inline src/app/space-wizard/space-wizard.component.html]: Use name/for with the label, then it can be anywhere

**Pull request merged**